casper/src/test/scala/coop/rchain/casper/HashSetCasperTest.scala

```diff
@@ -46,6 +49,29 @@ class HashSetCasperTest extends FlatSpec with Matchers {
     node.tearDown()
   }
 
+  it should "not allow multiple threads to process the same block" in {
+    val scheduler          = Scheduler.fixedPool("two-threads", 3)
+    val (casperT, cleanUp) = TaskCasper(validatorKeys.head, genesis)(scheduler)
```

Comment:
How testing something defined only for this test gives us the confidence that the thing used in the production code runs correctly as well?

Reply:
The type we use in production is just `Task` lifted into an `EitherT`, so using `Task` here is enough to test what we need.